adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/CoverageRDD.scala

```diff
@@ -0,0 +1,225 @@
+        if (condensed.map(r => ReferenceRegion(r)).filter(_.overlaps(ReferenceRegion(lastCoverage))).isEmpty) {
+          lastCoverage :: condensed
+        } else condensed
+      nextCondensed.toIterator
+    } else {
+      val cov: Coverage = iter.next
+      val rr = ReferenceRegion(cov)
+      val lastRegion = ReferenceRegion(lastCoverage)
+      val (nextCoverage: Coverage, nextCondensed: List[Coverage]) =
+        if (rr.isAdjacent(lastRegion) && lastCoverage.count == cov.count) {
+          (Coverage(rr.merge(lastRegion), lastCoverage.count), condensed)
+        } else {
+          (cov, (lastCoverage) :: condensed)
+        }
+      collapse(iter, nextCoverage, nextCondensed)
+    }
+  }
+
+  /**
+   * @return Returns a FeatureRDD from CoverageRDD
+   */
+  def toFeatureRDD: FeatureRDD = {
+    val featureRdd = rdd.map(_.toFeature)
+    FeatureRDD(featureRdd, sequences)
+  }
+
+  /**
+   * Gets coverage overlapping specified ReferenceRegion. For large, ReferenceRegions,
+   * base pairs per bin (bpPerBin) can be specified to bin together ReferenceRegions of
+   * equal size. The coverage of each bin is coverage of the first base pair in that bin.
+   *
+   * @param bpPerBin base pairs per bin, number of bases to combine to one bin
+   * @return RDD of Coverage Records
+   */
+  def coverage(bpPerBin: Int = 1): CoverageRDD = {
+
+    val flattened = this.flatten
+
+    if (bpPerBin == 1) {
+      flattened // no binning, return raw results
+    } else {
+      // subtract region.start to shift mod to start of ReferenceRegion
+      val newRDD = flattened.rdd.filter(r => r.start % bpPerBin == 0)
+      flattened.replaceRdd(newRDD)
+    }
+  }
+
+  /**
+   * Gets coverage overlapping specified ReferenceRegion. For large, ReferenceRegions,
+   * base pairs per bin (bpPerBin) can be specified to bin together ReferenceRegions of
+   * equal size. The coverage of each bin is the mean coverage over all base pairs in that bin.
+   *
+   * @param bpPerBin base pairs per bin, number of bases to combine to one bin
+   * @return RDD of Coverage Records
+   */
+  def aggregatedCoverage(bpPerBin: Int = 1): CoverageRDD = {
+
+    val flattened = this.flatten
+
+    def reduceFn(a: (Double, Int), b: (Double, Int)): (Double, Int) = {
+      (a._1 + b._1, a._2 + b._2)
+    }
+
+    if (bpPerBin == 1) {
+      flattened // no binning, return raw results
+    } else {
+      val newRDD = flattened.rdd
+        .keyBy(r => {
+          // key coverage by binning start site mod bpPerbin
+          // subtract region.start to shift mod to start of ReferenceRegion
+          val start = r.start - (r.start % bpPerBin)
+          ReferenceRegion(r.contigName, start, start + bpPerBin)
+        }).mapValues(r => (r.count, 1))
+        .reduceByKey(reduceFn)
+        .map(r => {
+          // compute average coverage in bin
+          Coverage(r._1.referenceName, r._1.start, r._1.end, r._2._1 / r._2._2)
+        })
+      flattened.replaceRdd(newRDD)
```

Comment:
The way I look at the `replaceRdd` method is that it exists solely so that the `transform` method could be implemented in the `GenomicRDD` trait. Ideally, it's never called elsewhere. However, I don't think this intention is documented. I'll go back and doc this.